tools/benchmark.cr

```diff
@@ -0,0 +1,57 @@
+require "../src/pluto"
```

Comment:
I think we need to make usage a bit easier.

Options:

- (preferred) Use [shebangs](<https://en.wikipedia.org/wiki/Shebang_(Unix)>) and omit the extension from the file.
- Add a comment at the top explaining how to run the script.

Btw, I think the command should be `crystal run --no-debug --release tools/benchmark.cr`.

Reply:
Also, `--release`! I thought my numbers seemed off. Like, my machine isn't the newest or fastest, but 5x slower than your results seemed way off. With these flags, I'm seeing similar numbers as you:

|                            | Time | Memory  |
| -------------------------- | ---- | ------- |
| bilinear_resize!(640, 480) | 12ms | 1.17MiB |
|           brightness!(1.4) | 2ms  | 0B      |
|              box_blur!(10) | 7ms  | 600kiB  |
| channel_swap!(:red, :blue) | 0ms  | 0B      |
|             contrast!(128) | 2ms  | 0B      |
|         gaussian_blur!(10) | 23ms | 1.75MiB |
|       horizontal_blur!(10) | 3ms  | 300kiB  |
|         vertical_blur!(10) | 4ms  | 300kiB  |

Total Time  : 53ms
Total Memory: 4.1MiB

Reply:
> How do you feel about makefiles? That's usually my go to for quick and easily accessible operations - make build, make test, and make run being common ones, with one off scripts like make benchmark too. They have a... mixed reputation in the dev community, I feel though.

I don't like Make, but I successfully use [Task](https://taskfile.dev/) in one of my projects. It is also used in some bigger projects: https://github.com/go-task/task/discussions/751.

However, an extra dependency is probably not worth it in this case anyway.

If you have better alternatives in mind, I can consider them in the future.

Reply:
Makes sense, and agreed, another tool dependency would be cumbersome. That's one of the few advantages that make gets - it's _everywhere_. This project is simple enough that it's not too bad to keep using the base crystal commands as they are :+1: